delete.go

```diff
@@ -58,24 +59,28 @@ status of "ubuntu01" as "stopped" the following will delete resources held for
 			}
 			return err
 		}
-		// When --force is given, we kill all container processes and
-		// then destroy the container. This is done even for a stopped
-		// container, because (in case it does not have its own PID
-		// namespace) there may be some leftover processes in the
-		// container's cgroup.
-		if force {
-			return killContainer(container)
-		}
+
 		s, err := container.Status()
 		if err != nil {
 			return err
 		}
 		switch s {
 		case libcontainer.Stopped:
+			// For a stopped container, because (in case it does not have
+			// its own PID namespace) there may be some leftover processes
+			// in the container's cgroup.
+			if !container.Config().Namespaces.IsPrivate(configs.NEWPID) {
+				return killContainer(container)
+			}
 			return container.Destroy()
 		case libcontainer.Created:
 			return killContainer(container)
 		default:
+			// When --force is given, we kill all container processes and
+			// then destroy the container.
+			if force {
+				return killContainer(container)
+			}
```

Comment:
Unless I'm missing something, commit description doesn't describe this change.

I also remember breaking something than I moved this code around. Ah! It was commit 29283bb7. I hope this change won't result in some sort of regression :)

Reply:
Ah, OK, you said in the commit message that you don't want to wait in case of a stopped container.